gapis/api/service.proto

```diff
@@ -238,3 +238,127 @@ message CubemapLevel {
 	image.Info negative_z = 5;
 	image.Info positive_z = 6;
 }
+
+message Metrics {
+  oneof metrics {
+    MemoryBreakdown memory_breakdown = 1;
+  }
+}
+
+// The description of the memory layout of the API state
+message MemoryBreakdown {
+  // The API ID used for this call.
+  path.API api = 1;
+  // The memory allocations that are currently in use.
+  repeated MemoryAllocation allocations = 2;
+  // The index to get the constants names for MemoryAllocation `flags`.
+  // A value of -1 indicates no flag names should be fetched.
+  int32 allocation_flags_index = 3;
+}
+
+// A single memory allocation
+message MemoryAllocation {
+  // The device this allocation was made on
+  uint64 device = 1;
+  // The memory type this allocation is from
+  uint32 memory_type = 2;
+  // Flags describing properties of the allocation, e.g. whether it's device
+  // local, whether it's host visible, etc.
+  uint32 flags = 3;
+  // Constant index for looking up the flag values with the path.ConstantSet endpoint
```

Comment:
Of course, I now see it is mentioned here. I of course didn't look here. :)

Reply:
Although now that you point this out I think this field is duplicating the work of `allocation_flags_index` in `MemoryBreakdown`.  That should be fixed.